core/src/main/java/com/linecorp/armeria/client/ArrayBasedEventLoopState.java

```diff
@@ -39,15 +38,18 @@ final class ArrayBasedEventLoopState extends AbstractEventLoopState {
         if (eventLoops.size() == maxNumEventLoops) {
             // We use all event loops so initialize early.
             init(0);
+        } else {
+            init(scheduler().acquisitionStartIndex(maxNumEventLoops));
         }
     }
 
-    private void init(int acquisitionStartIndex) {
-        acquisitionIndex = new EventLoopAcquisitionIndex(
-                acquisitionStartIndex, maxNumEventLoops, eventLoops().size());
-        for (int i = 0; i < maxNumEventLoops; ++i) {
-            push(new Entry(this, eventLoops().get(acquisitionIndex.nextAcquirableIndex()), i));
-        }
+    private void init(final int acquisitionStartIndex) {
+        final int initialEventLoopOffset = ThreadLocalRandom.current().nextInt(maxNumEventLoops);
+        final Function<Integer, Integer> acquisitionIndex =
+                offset -> acquisitionStartIndex + (initialEventLoopOffset + offset) % maxNumEventLoops;
+        entries.addAll(IntStream.rangeClosed(0, maxNumEventLoops - 1)
+                                .mapToObj(i -> new Entry(this, eventLoops().get(acquisitionIndex.apply(i)), i))
+                                .collect(Collectors.toList()));
```

Comment:
How about just using for loop?
```suggestion
        for (int i = 0; i < maxNumEventLoops; i++) {
            final int nextIndex = (acquisitionStartIndex + (initialEventLoopOffset + i) % maxNumEventLoops) %
                                  eventLoops().size();
            entries.add(new Entry(this, eventLoops().get(nextIndex), i));
        }
```